testing/test_runner.py

```diff
@@ -471,6 +471,15 @@ def test_importorskip_imports_last_module_part():
     ospath = pytest.importorskip("os.path")
     assert os.path == ospath
 
+def test_importorskip_dev_module():
+    mod = py.std.types.ModuleType("mockmodule")
+    mod.__version__ = '0.13.0.dev-43290'
+    sys.modules['mockmodule'] = mod
```

Comment:
Instead of setting into `sys.modules` directly, please do this instead:
```python
monkeypatch.setitem(sys.modules, 'mockmodule', mod)
```
This will ensure `mockmodule` will be removed from `sys.modules` up at the end of the test: